flink-core/src/test/java/org/apache/flink/api/common/typeutils/SerializerTestBase.java

```diff
@@ -409,9 +402,26 @@ else if (should instanceof Throwable) {
 			assertEquals(((Throwable)should).getMessage(), ((Throwable)is).getMessage());
 		}
 		else {
+			// if toString() was overridden we must check it
+			if (!should.toString().equals(should.getClass().getName() +
+										  "@" + Integer.toHexString(should.hashCode()))) {
+				assertEquals(message, should.toString(), is.toString());
+			}
 			assertEquals(message,  should, is);
 		}
 	}
+
+	protected void checkClass(String message, Class<?> type, Class<?> instanceType) {
```

Comment:
I don't understand the logic of this method. Why not simply compare the classes?